src/sentry/incidents/models/alert_rule.py

```diff
@@ -223,6 +223,20 @@ def conditionally_subscribe_project_to_alert_rules(
             )
         return []
 
+    def get_for_metrics(
+        self, organization: Organization, metric_mris: list[str]
+    ) -> BaseQuerySet[AlertRule]:
+        """
+        Fetches AlertRules associated with the metric MRIs
+        """
+
+        alert_query = Q()
+        for metric_mri in metric_mris:
+            alert_query |= Q(snuba_query__aggregate__contains=metric_mri)
+
+        queryset = self.filter(organization=organization).filter(alert_query)
+        return queryset
```

Comment:
nit: I'd expect this to be in `MetricsExtractionRule` since we only need this for Metric Extraction Rules, unless there is some other use case where we would want to query "which MRI is this specific alert depending on?", but then it would make sense to rewrite this into a method like `def get_metric(self) -> MRI`. 
Same for the widgets btw.

Reply:
Actually here we are querying: "give me all alerts that depend on this MRIs", which is basically a special type of filtering on `AlertRule` queryset, and that is why I have added it as a method on the `AlertRuleManager`.

I have no strong opinions here, but from my point of view, these complex filtering options should be in the custom Manager, that's why the Managers exist in Django